File: Smallest Multiple.java
import java.util.*;
public class Cp{
	static Scanner sc = new Scanner(System.in);
	public static void main(String[]args){
	int res = 1;
	for(int i = 2 ; i <= 20 ; i++){
		res = lcm(res, i);
	}
	System.out.print(res);
	
}
static int lcm(int x , int y){
	return x * (y/gcd(x,y));
}
static int gcd(int x, int y){
	while(x != 0 ){
	int z = x; 
	x = y % x;
	y = z;
}
	return y;
}
}

File: Smallest Multiple.cpp
#include <bits/stdc++.h>
#define lli long long int
using namespace std;
lli gcd(lli a, lli b){
	if(b==0)
		return a;
	return gcd(b, a%b);
}
lli lcm(lli a, lli b){
	return (a*b) /(gcd(a,b));
}
int main(){
	lli n;
	cin >>n;
	lli res =1;
	for(lli i =2 ; i<= n; i++)
		res = lcm(res, i);
	cout << res;
}
